plugin/trino-hive/src/test/java/io/trino/plugin/hive/AbstractTestHiveFileSystem.java

```diff
@@ -400,6 +406,65 @@ public void testRename()
         fs.delete(basePath, true);
     }
 
+    @Test
+    public void testFileIteratorListing()
+            throws Exception
+    {
+        Table.Builder tableBuilder = Table.builder()
+                .setDatabaseName(table.getSchemaName())
+                .setTableName(table.getTableName())
+                .setDataColumns(ImmutableList.of(new Column("one", HIVE_LONG, Optional.empty())))
+                .setPartitionColumns(ImmutableList.of())
+                .setOwner(Optional.empty())
+                .setTableType("fake");
+        tableBuilder.getStorageBuilder()
+                .setStorageFormat(StorageFormat.fromHiveStorageFormat(HiveStorageFormat.CSV));
+        Table fakeTable = tableBuilder.build();
+
+        Path basePath = new Path(getBasePath(), "test-file-iterator-listing");
+        FileSystem fs = hdfsEnvironment.getFileSystem(TESTING_CONTEXT, basePath);
+        fs.mkdirs(basePath);
+
+        // create file in hidden folder
+        Path hiddenFile = new Path(new Path(basePath, ".hidden"), "hidden-file.txt");
+        fs.create(hiddenFile);
```

Comment:
I tried to run locally the new test and it was failing because no file has been created.

You may want to use `fs.createNewFile` to actually close the output stream and trigger the creation of the new file.

Reply:
You're right, I completely blanked and forgot that `fs.create` just opens the output stream. I switched all of these calls to `fs.createNewFile`.